components/reddit/reddit.app.js

```diff
@@ -0,0 +1,56 @@
+const axios = require("axios");
+const get = require("lodash.get");
+module.exports = {
+  type: "app",
+  app: "reddit",
+  propDefinitions: {
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch.",
+    }
+  },
+  methods: {
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    _apiUrl() {
+      return `https://oauth.reddit.com`;
+    },
+    async _makeRequest(opts) {
+      if (!opts.headers) opts.headers = {};
+      opts.headers.authorization = `Bearer ${this._accessToken()}`;
+      opts.headers["user-agent"] = "@PipedreamHQ/pipedream v0.1";
+      const { path } = opts;
+      delete opts.path;
+      opts.url = `${this._apiUrl()}${path[0] === "/" ? "" : "/"}${path}`;
+      return (await axios(opts)).data;
+    },
+    wereLinksPulled(reddit_things) {
+      const links = get(reddit_things, "data.children");
+      return links && links.length;
+    },
+    async getNewHotSubredditPosts(subreddit, g, show, sr_detail, limit = 100) {
+      let params = new Object();
```

Comment:
`params = {}` is more common, I'd recommend that.

You can also change to `const params`. You can add / modify properties of an object defined as a constant, you just can't overwrite the `params` variable itself.

Reply:
Got it. 
So most of this coding is legacy from Java C# where everything is so strongly typed, I don't recall adding/modying properties once the object was created. Same for new OBject().
I made the change and will take the recommendations into account for future instances when declaring a constant. All right.